config/config.go

```diff
@@ -111,43 +111,85 @@ func readConfig(bytes []byte) error {
 
 func validateServiceParameters(params serviceConfig) error {
 	if params.Port < 0 || params.Port > 65535 {
-		return fmt.Errorf("Invalid port: %d (must be 0-65535)", params.Port)
+		return InvalidServiceConfigError{
+			Message: fmt.Sprintf("Invalid port: %d (must be 0-65535)", params.Port),
+		}
 	}
 	if params.MaxConnections <= 0 {
-		return fmt.Errorf("Invalid max_connections: %d (must be positive)",
-			params.MaxConnections)
+		return InvalidServiceConfigError{
+			Message: fmt.Sprintf("Invalid max_connections: %d (must be positive)",
+				params.MaxConnections),
+		}
 	}
 	if params.Endpoint != "" {
-		if _, endpointFound := Endpoints[params.Endpoint]; !endpointFound {
-			return fmt.Errorf("Invalid service endpoint: %s", params.Endpoint)
+		if _, found := Endpoints[params.Endpoint]; !found {
+			return InvalidServiceConfigError{
+				Message: fmt.Sprintf("Invalid endpoint: %s", params.Endpoint),
+			}
 		}
 	}
 	if params.PollInterval <= 0 {
-		return fmt.Errorf("Non-positive poll interval specified: (%d s)",
-			params.PollInterval)
+		return InvalidServiceConfigError{
+			Message: fmt.Sprintf("Non-positive poll interval specified: (%d s)",
+				params.PollInterval),
+		}
 	}
 	if params.DeleteAfter <= 0 {
-		return fmt.Errorf("Non-positive task deletion period specified: (%d h)",
-			params.DeleteAfter)
+		return InvalidServiceConfigError{
+			Message: fmt.Sprintf("Non-positive task deletion period specified: (%d h)",
+				params.DeleteAfter),
+		}
 	}
 	return nil
 }
 
 func validateEndpoints(endpoints map[string]endpointConfig) error {
-	for label, endpoint := range endpoints {
+	for name, endpoint := range endpoints {
 		if endpoint.Id.String() == "" { // invalid endpoint UUID
-			return fmt.Errorf("Invalid UUID specified for endpoint '%s'", label)
+			return InvalidEndpointConfigError{
+				Endpoint: name,
+				Message:  "Invalid UUID",
+			}
 		} else if endpoint.Provider == "" { // no provider given
-			return fmt.Errorf("No provider specified for endpoint '%s'", label)
+			return InvalidEndpointConfigError{
```

Comment:
minor: no need for `else if` after return -- `if` alone is fine

Reply:
Actually, this one needs the `else if`, because there are two things to check for each config item, and if they both check out for all items, a `nil` error is returned. Or have I misunderstood something?